conan/tools/cmake/base.py

```diff
@@ -143,6 +148,15 @@ def __init__(self, conanfile, **kwargs):
 
         self.build_type = None
 
+        self.find_package_prefer_config = \
+            conanfile.conf["tools.cmake.cmaketoolchain"].find_package_prefer_config
+        if self.find_package_prefer_config is not None:
```

Comment:
This is not opting-out, it is opting-in please recall my comment:

> Sounds good, lets add a [conf] to opt-out this behavior, I'd suggest something like


Then simplify code, as you said they are always strings, so go for a oneliner:

```python
if self.find_package_prefer_config is not None:
    self.find_package_prefer_config = "OFF" if self.find_package_prefer_config.lower() in ("false", "0", "off") else "ON"
```

Reply:
I am not sure I understand. this is opt-out, the behavior matches CMake exactly:
- conf not defined -> CMAKE_FIND_PACKAGE_PREFER_CONFIG not define
- conf=True -> CMAKE_FIND_PACKAGE_PREFER_CONFIG=ON
- conf=False -> CMAKE_FIND_PACKAGE_PREFER_CONFIG=OFF

Reply:
The whole idea was to have it ON by default so inexperienced users using `conanfile.txt` without any `conf` sections have a friendly dev env. Advanced users may want to disable it if they so desire.

As a side note, still there are many inexperienced users out there both for Conan and CMake. Doing things that we consider trivial is not easy for them at all. For the last two days, I am remotely helping one of them to enable `mp-units` support via `conanfile.txt` even though I thought that my docs are clear: https://mpusz.github.io/units/usage.html#conan-cmake-release. Today I finally saw that person's CMake file and it turned out to be:

![image](https://user-images.githubusercontent.com/506260/111635869-3d98df00-87f8-11eb-9ae0-47d6221fd6de.png)

Reply:
okay, I added the ON as default again. there was a confusion because initially I implemented ON as default, then others said it should be controlled by conf instead. hopefully the current behavior makes sense:
- conf not defined -> CMAKE_FIND_PACKAGE_PREFER_CONFIG=ON
- conf=True -> CMAKE_FIND_PACKAGE_PREFER_CONFIG=ON
- conf=False -> CMAKE_FIND_PACKAGE_PREFER_CONFIG=OFF

Reply:
Actually, in this case, a user correctly applied `CONFIG REQUIRED` as suggested in my docs. The problem was with the last two lines 😉

Reply:
@SSE4 my request was:

> Sounds good, lets add a [conf] to opt-out this behavior, I'd suggest something like:

A config required to **opt-out**. That means that by default it is ON, and you need to do something proactively to opt-out of that feature. If you need to define a conf for activating the feature, it is not an opt-out feature, it is an opt-in feature.